### PR TITLE
PWGHF: Define version 001 of track-index skim creator tables to fix Run2 validation

### PR DESCRIPTION
**Pull request by fgrosa**
@vkucera this PR fixes the the bug spotted by @jgrosseo due to the introduction of the collision id in the tables produced in the track-index-skim-creator as a consequence of the development for the treatment of the ambiguous tracks.
In fact, in the Run2 converted data, the collision id is not present, hence it has to be added on the fly.

### CODE REVIEW

File: PWGHF/DataModel/CandidateReconstructionTables.h [hunk 1]
```diff
@@ -125,12 +125,20 @@ DECLARE_SOA_COLUMN(FlagDsToKKPi, flagDsToKKPi, uint8_t);         //!
 DECLARE_SOA_COLUMN(FlagXicToPKPi, flagXicToPKPi, uint8_t);       //!
 } // namespace hf_track_index
 
-DECLARE_SOA_TABLE(Hf2Prongs, "AOD", "HF2PRONG", //! Table for HF 2 prong candidates
+DECLARE_SOA_TABLE(Hf2Prongs_000, "AOD", "HF2PRONG", //! Table for HF 2 prong candidates
```
Comment (jgrosseo) [inline PWGHF/DataModel/CandidateReconstructionTables.h:128]:
```suggestion
DECLARE_SOA_TABLE(Hf2Prongs_000, "AOD", "HF2PRONG", //! Table for HF 2 prong candidates (Run 2 converted format)
```

File: PWGHF/DataModel/CandidateReconstructionTables.h [hunk 2]
```diff
@@ -125,12 +125,20 @@ DECLARE_SOA_COLUMN(FlagDsToKKPi, flagDsToKKPi, uint8_t);         //!
...
                   o2::soa::Index<>,
-                  hf_track_association::CollisionId,
                   hf_track_index::Prong0Id,
                   hf_track_index::Prong1Id,
                   hf_track_index::HFflag);
+
+DECLARE_SOA_TABLE_VERSIONED(Hf2Prongs_001, "AOD", "HF2PRONG", 1, //! Table for HF 2 prong candidates
```
Comment (jgrosseo) [inline PWGHF/DataModel/CandidateReconstructionTables.h:134]:
```suggestion
DECLARE_SOA_TABLE_VERSIONED(Hf2Prongs_001, "AOD", "HF2PRONG", 1, //! Table for HF 2 prong candidates (Run 3 format)
```

**Comment by jgrosseo** [comment]
Thanks a lot!

@vkucera if we could merge this tonight, this would be great!

File: PWGHF/DataModel/CandidateReconstructionTables.h [hunk 3]
```diff
@@ -140,13 +148,22 @@ DECLARE_SOA_TABLE(HfCascades, "AOD", "HFCASCADE", //! Table for HF candidates wi
                   hf_track_index::V0Id);
 using HfCascade = HfCascades::iterator;
 
-DECLARE_SOA_TABLE(Hf3Prongs, "AOD", "HF3PRONG", //! Table for HF 3 prong candidates
+DECLARE_SOA_TABLE(Hf3Prongs_000, "AOD", "HF3PRONG", //! Table for HF 3 prong candidates
```
Comment (jgrosseo) [inline PWGHF/DataModel/CandidateReconstructionTables.h:151]:
```suggestion
DECLARE_SOA_TABLE(Hf3Prongs_000, "AOD", "HF3PRONG", //! Table for HF 3 prong candidates (Run 2 converted format)
```

File: PWGHF/DataModel/CandidateReconstructionTables.h [hunk 4]
```diff
@@ -140,13 +148,22 @@ DECLARE_SOA_TABLE(HfCascades, "AOD", "HFCASCADE", //! Table for HF candidates wi
...
                   o2::soa::Index<>,
-                  hf_track_association::CollisionId,
                   hf_track_index::Prong0Id,
                   hf_track_index::Prong1Id,
                   hf_track_index::Prong2Id,
                   hf_track_index::HFflag);
+
+DECLARE_SOA_TABLE_VERSIONED(Hf3Prongs_001, "AOD", "HF3PRONG", 1, //! Table for HF 3 prong candidates
```
Comment (jgrosseo) [inline PWGHF/DataModel/CandidateReconstructionTables.h:158]:
```suggestion
DECLARE_SOA_TABLE_VERSIONED(Hf3Prongs_001, "AOD", "HF3PRONG", 1, //! Table for HF 3 prong candidates (Run 3 format)
```

**Review by vkucera — APPROVED** [review]

**Comment by vkucera** [comment]
Thanks @fgrosa @jgrosseo

**Pull request merged**